testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentUp.java

```diff
@@ -139,6 +144,14 @@ public Integer call()
                 builder.removeContainers(container -> isTrinoContainer(container.getLogicalName()));
             }
 
+            if (debug) {
+                builder.configureContainers(container -> {
+                    if (isTrinoContainer(container.getLogicalName())) {
+                        container.withStartupTimeout(NO_TIMEOUT);
```

Comment:
Are we trying to debug on why it fails or ?

Reply:
yes

Reply:
This works for both coordinator and workers. If they are very long in the debug mode - container will not fail because of timeout exception.